### PR TITLE
KAFKA-7080 and KAFKA-7222: Cleanup overlapping KIP changes

### PR DESCRIPTION
**Pull request by mjsax**
- KIP-319 and KIP-328 overlap and we can remove non-released deprecates methods
- add upgrade docs for KIP-319

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -115,6 +115,18 @@ <h3><a id="streams_api_changes_210" href="#streams_api_changes_210">Streams API
         For more details, see <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-358%3A+Migrate+Streams+API+to+Duration+instead+of+long+ms+times">KIP-358</a>.
     </p>
 
+    <p>
+        We deprecated the notion of segments in window stores as those are intended to be an implementation details.
+        Thus, method <code>Windows#segments()</code> and variable <code>Windows#segments</code> were deprecated.
+        If you implement custom windows, you should updated your code accordingly.
```
Comment (bbejeck) [inline docs/streams/upgrade-guide.html]: nit: `updated` -> `update`

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -115,6 +115,18 @@ <h3><a id="streams_api_changes_210" href="#streams_api_changes_210">Streams API
...
+        Similarly, <code>WindowBytesStoreSupplier#segments()</code> was deprecated and replaced with <code>WindowBytesStoreSupplier#segmentInterval()</code>.
+        If you implement custom window store, you need to updated your code accordingly.
```
Comment (bbejeck) [inline docs/streams/upgrade-guide.html]: same here

**Review by bbejeck — APPROVED** [review]
LGTM just two very minor comments

File: streams/src/main/java/org/apache/kafka/streams/state/Stores.java [hunk 3]
```diff
@@ -269,7 +235,9 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
      *                          windowed data's entire life cycle, from window-start through window-end,
      *                          and for the entire grace period.
      * @return an instance of a {@link  SessionBytesStoreSupplier}
+     * @deprecated since 2.1 Use {@link Stores#persistentSessionStore(String, Duration)} instead
      */
+    @Deprecated
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/Stores.java:240]: This is part of KIP-358 but slipped in the PR

File: streams/src/main/java/org/apache/kafka/streams/kstream/Windows.java [hunk 4]
```diff
@@ -83,21 +83,6 @@ public long maintainMs() {
...
     }
 
-    /**
-     * Return the segment interval in milliseconds.
-     *
-     * @return the segment interval
-     * @deprecated since 2.1. Instead, directly configure the segment interval in a store supplier and use {@link Materialized#as(WindowBytesStoreSupplier)}.
-     */
-    @Deprecated
-    public long segmentInterval() {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Windows.java:93]: added in KIP-319, deprecated in KIP-328

File: streams/src/main/java/org/apache/kafka/streams/state/Stores.java [hunk 5]
```diff
@@ -178,28 +178,6 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
...
-     *                              Note that the retention period must be at least long enough to contain the
-     *                              windowed data's entire life cycle, from window-start through window-end,
-     *                              and for the entire grace period.
-     * @param windowSize            size of the windows (cannot be negative)
-     * @param retainDuplicates      whether or not to retain duplicates.
-     * @return an instance of {@link WindowBytesStoreSupplier}
-     * @deprecated Use {@link #persistentWindowStore(String, Duration, Duration, boolean)} instead
-     */
-    @Deprecated
-    public static WindowBytesStoreSupplier persistentWindowStore(final String name,
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/Stores.java:194]: added in KIP-319, deprecated in KIP-358

File: streams/src/main/java/org/apache/kafka/streams/state/Stores.java [hunk 6]
```diff
@@ -220,28 +198,16 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
...
-     *                              windowed data's entire life cycle, from window-start through window-end,
-     *                              and for the entire grace period.
-     * @param segmentInterval       size of segments in ms (cannot be negative)
-     * @param windowSize            size of the windows (cannot be negative)
-     * @param retainDuplicates      whether or not to retain duplicates.
-     * @return an instance of {@link WindowBytesStoreSupplier}
-     * @deprecated Use {@link #persistentWindowStore(String, Duration, Duration, boolean)} instead
-     */
-    @Deprecated
-    public static WindowBytesStoreSupplier persistentWindowStore(final String name,
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/Stores.java:240]: added in KIP-319 -- should have bee deprecated with KIP-328.

**Review by guozhangwang — APPROVED** [review]
@mjsax please feel free to merge after rebase.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.1`.

**Comment by enothereska** [comment]
I think there is an error in materialize line 198 in org.apache.kafka.streams.kstream.internals.TimeWindowedKstreamImpl.

**Comment by ewencp** [comment]
@enothereska There's a follow up already in #5806 that I think addresses your concern

**Pull request merged**